src/lib.rs

```diff
@@ -1138,6 +1145,16 @@ mod tests {
         assert_tokens(&sig.readable(), &[Token::BorrowedStr(SIG_STR)]);
     }
 
+    #[cfg(feature = "global-context")]
+    #[test]
+    fn test_global_context() {
+        use super::SECP256K1;
+
+        let (sk, pk) = SECP256K1.generate_keypair(&mut thread_rng());
```

Comment:
`thread_rng()` requires `std`. Maybe just use a hardcoded key?

Reply:
The tests always have access to std AFAIR

Reply:
I have no idea. If yes, feel free to ignore my comment.

Reply:
Your remark reminded me of something different: I still need to add a test to travis with this feature activated. Imo I can just activate std in the test too.

Reply:
Sure, that works. But it isn't it a better test if it can be compiled with std on and off?

Reply:
Tbh, I don't care much about it either way, I can just change it. The main purpose was to see if I can use the global secp context both as a direct function argument and with `.function(…)` where it gets implicitly passed as self. So for me this is mostly a compile test to see that whatever wrapping `lazy-static` does is transparent.

Reply:
@real-or-random Do you think the current version is better?

Reply:
Yep, it was certainly not a crucial thing but I'd prefer the new one.